modules/rosetta/src/main/scala/org/tessellation/rosetta/domain/Amount.scala

```diff
@@ -1,31 +1,60 @@
 package org.tessellation.rosetta.domain
 
 import org.tessellation.ext.derevo.magnoliaCustomizable.snakeCaseConfiguration
-import org.tessellation.rosetta.domain.currency.Currency
+import org.tessellation.rosetta.domain.currency.{Currency, DAG}
+import org.tessellation.schema.transaction.{TransactionAmount, TransactionFee}
 
-import derevo.cats.eqv
+import derevo.cats.{eqv, show}
 import derevo.circe.magnolia._
 import derevo.derive
 import eu.timepit.refined.api.Refined
 import eu.timepit.refined.auto._
 import eu.timepit.refined.boolean.Not
 import eu.timepit.refined.cats._
 import eu.timepit.refined.generic.Equal
+import eu.timepit.refined.numeric.{NonNegative, Positive}
+import eu.timepit.refined.refineV
 import io.circe.refined._
 import io.estatico.newtype.macros.newtype
 
+// TODO should we rename Amount to NonZeroAmount ?
+//  it would be clearer and not clash with schema.balance.Amount
+
+// TODO should we unpack Amount.value and make it Refined[Long, NonZero] where type NonZero = Not[Equal[0L]] ?
+//  I'm not clear what we gain by the extra wrapping other than more work by callers
```

Comment:
> // TODO should we unpack Amount.value and make it Refined[Long, NonZero] where type NonZero = Not[Equal[0L]] ?
//  I'm not clear what we gain by the extra wrapping other than more work by callers

I didn't understand what you mean. where exactly you want to unpack it? can you show an exemplary code?

Reply:
@marcinwadon 
I'm asking if it makes sense to change
```
case class Amount(value: AmountValue, currency: Currency)
```
to
```
case class Amount(value: AmountValueRefined, currency: Currency)
```

Reply:
Well, IMO it makes sense to use newtypes. It delegates the implementation detail to a new type accessible by an expressive name, and you cannot pass a different value by mistake. Refined types are not newtypes.

Reply:
@marcinwadon There are already examples of this in the code base (e.g. _TransactionAmount_). What do we gain but nesting yet another type to represent the value that we aren't getting with Refined types?

Reply:
This is exactly the same case as using `TransactionAmount(value: NonNeglong)` and then `Foo(..., amount: TransactionAmount)` or using `Foo(..., amount: NonNegLong)`